src/Simulator/IncrementalRedstoneSimulator/ObserverHandler.h

```diff
@@ -0,0 +1,106 @@
+
+#pragma once
+
+#include "RedstoneHandler.h"
+#include "../../Blocks/BlockObserver.h"
+
+
+
+
+
+class cObserverHandler : public cRedstoneHandler
+{
+public:
+
+	inline static bool IsOn(NIBBLETYPE a_Meta)
+	{
+		return (a_Meta & 0x8) == 0x8;
+	}
+
+	static bool ShouldPowerOn(cWorld & a_World, const Vector3i & a_Position, NIBBLETYPE a_Meta, cIncrementalRedstoneSimulatorChunkData * a_Data)
+	{
+		BLOCKTYPE BlockType;
+		NIBBLETYPE BlockMeta;
+		if (!a_World.GetBlockTypeMeta(a_Position + cBlockObserverHandler::GetObservingFaceOffset(a_Meta), BlockType, BlockMeta))
+		{
+			return false;
+		}
+
+		// Cache the last seen block type and meta in the power data for this position
+		auto ಠ_ಠ = PoweringData(BlockType, BlockMeta);
```

Comment:
```suggestion
		auto kukačka = PoweringData(BlockType, BlockMeta);
```
:P
About the same readability.